### PR TITLE
[master] add disk pool commit to userid and show userid with G if memory is ju…

### PR DESCRIPTION
**Pull request by wngzhe**
code change:
1.  if memory size is Integer multiple of GB ，userid shows as xG
2. add diskpool information  to userid
3. change unittest cases

### CODE REVIEW

File: zvmsdk/smtclient.py [hunk 1]
```diff
@@ -674,6 +679,16 @@ def create_vm(self, userid, cpu, memory, disk_list, profile,
...
+        disk_pool_info = "DISK_POOL " + CONF.zvm.disk_pool if CONF.zvm.disk_pool else None
+        if len(disk_list) == 1:
+            disk = disk_list[0]
+            if 'format' in disk and disk['format'].lower() == 'swap':
+                disk_pool_info = disk.get('disk_pool') or CONF.zvm.disk_pool
+        if disk_pool_info:
+            if comment_list is not None:
+                comment_list.append(disk_pool_info)
+            else:
+                comment_list = [disk_pool_info]
```
Comment (SeanHQF) [inline zvmsdk/smtclient.py:691]: any test result for this?

**Comment by wngzhe** [comment]
test result：
[root@rhel-84-115-120-reg-upgrade-test-compute ~]# smcli Image_Query_DM  -T  WGNZ001f
USER WGNZ001F LBYONLY 4G 64G G
INCLUDE ZCCDFLT
COMMAND SET VCONFIG MODE LINUX
COMMAND DEFINE CPU 00 TYPE IFL
COMMAND DEF STOR RESERVED 61440M
* DISK_POOL ECKD:ICICPL
IPL 0100
LOGONBY DEVCORE1
MACHINE ESA 32
SHARE RELATIVE 100
MDISK 0100 3390 511479 14564 T60200 MR
*DVHOPT LNK0 LOG1 RCM1 SMS0 NPW1 LNGAMENG PWC20230620 CRCXX

**Comment by wngzhe** [comment]
[root@rhel-84-115-120-reg-upgrade-test-compute ~]# smcli Image_Query_DM  -T WGNZ0020
USER WGNZ0020 LBYONLY 4099M 64G G
INCLUDE ZCCDFLT
COMMAND SET VCONFIG MODE LINUX
COMMAND DEFINE CPU 00 TYPE IFL
COMMAND DEF STOR RESERVED 61437M
* DISK_POOL ECKD:ICICPL
IPL 0100
LOGONBY DEVCORE1
MACHINE ESA 32
SHARE RELATIVE 100
NICDEF 1000 TYPE QDIO DEVICES 3 MACID 7241FC LAN SYSTEM VSICIC
MDISK 0100 3390 571904 14564 T60200 MR
*DVHOPT LNK0 LOG1 RCM1 SMS0 NPW1 LNGAMENG PWC20230626 CRCXX
[root@rhel-84-115-120-reg-upgrade-test-compute ~]#

File: zvmsdk/smtclient.py (hunk 1 above)
Comment (wngzhe) [reply zvmsdk/smtclient.py:691]: added in comment

**Review by SeanHQF — APPROVED** [review]

**Comment by SeanHQF** [comment]
@jichenjc @bjhuangr pls help review, thx

File: zvmsdk/smtclient.py [hunk 2]
```diff
@@ -674,6 +679,16 @@ def create_vm(self, userid, cpu, memory, disk_list, profile,
                 raise exception.SDKInvalidInputFormat(msg=errmsg)
             rd += ' --commandPcif %s' % pcif
 
+        disk_pool_info = "DISK_POOL " + CONF.zvm.disk_pool if CONF.zvm.disk_pool else None
```
[the remark this answers is not in the document]
Comment (wngzhe) [reply zvmsdk/smtclient.py:682]: disk_list may input as none ,so we try get disk_pool from "CONF.zvm.disk_pool"， first，
if the another configuration exited in disk_list, try get from it,
reference the  line 731 in this file.

[the remark this answers is not in the document]
Comment (jichenjc) [reply zvmsdk/smtclient.py:682]: didn't get your point, the purpose of the change is to make M=>G, and that's all?  why need update disk_pool list?

[the remark this answers is not in the document]
Comment (wngzhe) [reply zvmsdk/smtclient.py:682]: sorry for missing information, this pr also fixed the issue:https://github.ibm.com/zvc/planning/issues/9459
, these 2 issue code change in same method and both need change a lot of test cases ,so I use one pr.

Comment (jichenjc) [inline zvmsdk/smtclient.py:682]: >>>        disk_pool_info = "DISK_POOL " + CONF.zvm.disk_pool if CONF.zvm.disk_pool else None

please rename to `DASD GROUP`

Comment (jichenjc) [reply zvmsdk/smtclient.py:682]: you will put None if disk_list != 1?

**Review by jichenjc — COMMENTED** [review]
I suggest you split the change
such combination is not something helpful for review and more important
and hard to handle for future, e.g revert part of the code

File: zvmsdk/smtclient.py (hunk 2 above)
Comment (wngzhe) [reply zvmsdk/smtclient.py:682]: if disk_list != 1 and  CONF.zvm.disk_pool is none ,disk pool information  will not be added to comment,

**Comment by wngzhe** [comment]
> I suggest you split the change such combination is not something helpful for review and more important and hard to handle for future, e.g revert part of the code

ok, let me submitted 2 pr for this

**Comment by SeanHQF** [comment]
I think this can be closed. @wngzhe

**Pull request closed without merging**